account-integrations/safe/src/SafeWebAuthnPlugin.sol

```diff
@@ -67,35 +67,92 @@ contract SafeWebAuthnPlugin is BaseAccount {
         return _publicKey;
     }
 
+    struct LocalVarStruct {
+        bytes1 authenticatorDataFlagMask;
+        uint256 clientChallengeDataOffset;
+    }
+
     function _validateSignature(
         UserOperation calldata userOp,
         bytes32 userOpHash
     ) internal override returns (uint256 validationData) {
-        {
+        bytes calldata authenticatorData;
+        bytes calldata clientData;
+        uint256[2] calldata signature;
+        uint256[2] calldata pubKey;
+        LocalVarStruct memory s;
+
+        {            
+            // parse length of all fixed-length params (including length)
+            uint i = 0;
+            uint dataLen = 32;
+            uint256 paramLen = abi.decode(userOp.signature[i:i+dataLen], (uint256));
+            // Fixed-length params (bytes1, (uint256?), uint256, uint256[2], uint256[2]). Expect 8 slots = 256 bytes
+            i += dataLen; // advance index
+
+            // decode fixed length params (values to memory)
+            dataLen = 3 * 32; //lenFixedParams - 32; // -32 already read length
             (
-                bytes memory authenticatorData,
-                bytes1 authenticatorDataFlagMask,
-                bytes memory clientData,
-                uint256 clientChallengeDataOffset,
-                uint256[2] memory signature,
-                uint256[2] memory pubKey
+                s.authenticatorDataFlagMask,
+                , // some number
+                s.clientChallengeDataOffset
             ) = abi.decode(
-                    userOp.signature,
-                    (bytes, bytes1, bytes, uint256, uint256[2], uint256[2])
-                );
-
-            bool verified = FCL_WebAuthn.checkSignature(
-                authenticatorData,
-                authenticatorDataFlagMask,
-                clientData,
-                userOpHash,
-                clientChallengeDataOffset,
-                signature,
-                pubKey
+                userOp.signature[i:i+dataLen],
+                (
+                    bytes1,
+                    uint256, //not sure what is encoded here
+                    uint256
+                )
             );
-            if (!verified) return SIG_VALIDATION_FAILED;
-            return 0;
+            i += dataLen; // advance index
+
+
+            bytes calldata calldataLocation;
+            // load fixed length array params (pointers to calldata)
+            dataLen = 2 * 32;
+            calldataLocation = userOp.signature[i:i+dataLen];
+            assembly{
+                signature := calldataLocation.offset
```

Comment:
To make sure I'm understanding, we assign the offset to the signature variable and not the `calldataLocation` bytes because solidity expects it to point to the location in calldata instead of the whole value?

I'm taking this understanding from this snippet in the solidity docs:
> local variables that refer to statically-sized calldata arrays or calldata structs evaluate to the address of the variable in calldata, not the value itself.